src/item.ts

```diff
@@ -26,6 +26,7 @@ interface Fields extends Format {
     bylineHtml: Option<DocumentFragment>;
     publishDate: Option<Date>;
     mainImage: Option<ImageData>;
+    mainVideo: Option<VideoData>;
```

Comment:
Is it possible for an article to have *both* a main image and a main video? Is it possible for it to have neither? I'm guessing the second may happen (interactive header?) but the first won't?

To prevent invalid states, can we make use of a new type here?

```ts
const enum MainMediaKind {
    Image,
    Video,
}

type MainMedia = {
    kind: MainMediaKind.Image;
    ...videoData;
} | {
    kind: MainMediaKind.Video;
    ...imageData;
}

interface Fields extends Format {
    ...
    mainMedia: Option<MainMedia>;
    ...
}
```

Reply:
I think it's rare but it can be more than Images and videos. This series uses an Instagram embed 🙃 https://www.theguardian.com/books/booksblog/2020/may/18/tips-links-and-suggestions-what-are-you-reading-this-week.

Reply:
I've created this type and extended it to other article types standard, liveblog, video.